pkg/patterns/declarative/image.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package declarative
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative/pkg/manifest"
+)
+
+// ImageRegistryTransform modifies all Pods to use registry for the image source and adds the imagePullSecret
+func ImageRegistryTransform(registry, imagePullSecret string) ObjectTransform {
+	return func(c context.Context, o DeclarativeObject, m *manifest.Objects) error {
+		return applyImageRegistry(c, o, m, registry, imagePullSecret)
+	}
+}
+
+func applyImageRegistry(ctx context.Context, operatorObject DeclarativeObject, manifest *manifest.Objects, registry, secret string) error {
+	if registry == "" && secret == "" {
+		return nil
+	}
+	for _, manifestItem := range manifest.Items {
+		if manifestItem.Kind == "Deployment" || manifestItem.Kind == "DaemonSet" ||
+			manifestItem.Kind == "StatefulSet" || manifestItem.Kind == "Job" ||
+			manifestItem.Kind == "CronJob" {
```

Comment:
We might later want to extract this into a "visit images" helper.  But only when we need it next!